FirebaseDeviceStore/FirebaseDeviceStore.swift

```diff
@@ -123,8 +125,21 @@ public class FirebaseDeviceStore: NSObject, MessagingDelegate {
     private func deleteDevice(_ userId: String) {
         let docRef = userRef(userId);
 
+        // TODO: Rather than store a 'devices' array, you could store it as a map:
+        // devices: {
+        //   deviceId1: { },
+        //   deviceID2: { },
+        //   ...
+        // }
+        //
+        // And then to delete you don't need to use a transaction and you can just do
+        // docRef.updateData(FieldPath.of(self.DEVICES_FIELD, getDeviceId()), FieldValue.delete());
+
         firestore.runTransaction({ (transaction, errorPointer) -> Any? in
             let doc: DocumentSnapshot;
+            // TODO: Is this do/catch block necessary? I think if you just let the exception bubble
+            // up to Firestore, then it'll cancel the transaction and propagate the error
+            // automatically.
```

Comment:
Ahh, I see this comes straight from the docs example.  So feel free to ignore.

Reply:
Yes, I initially tried without catching the error, but it wouldn't compile so had to follow the example